ogr/utils.py

```diff
@@ -4,24 +4,76 @@
 import functools
 import logging
 import re
+import time
 from collections.abc import Iterable
 from re import Match
-from typing import Any, Callable, Optional, Union
+from typing import TYPE_CHECKING, Any, Callable, Optional, Union
 
 from urllib3.util import Retry
 
+if TYPE_CHECKING:
+    from urllib3.response import BaseHTTPResponse
+else:
+    # At runtime, BaseHTTPResponse doesn't exist, use HTTPResponse
```

Comment:
I'm sorry, I don't get this. How can it not exist? I can import it just fine in an interactive Python shell.